pkg/appmeshinject/inject.go

```diff
@@ -0,0 +1,197 @@
+package appmeshinject
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/k8s"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"strconv"
+	"strings"
+)
+
+const (
+	ecrSecret = "appmesh-ecr-secret"
+	// We don't want to make this configurable since users shouldn't rely on this
+	// feature to set a fsGroup for them. This feature is just to protect innocent
+	// users that are not aware of the limitation of iam-for-service-accounts:
+	// https://github.com/aws/amazon-eks-pod-identity-webhook/issues/8
+	// Users should set fsGroup on the pod spec directly if a specific fsGroup is desired.
+	defaultFSGroup int64 = 1337
+)
+
+var (
+	config    Config
+	injectLog = ctrl.Log.WithName("appmesh_inject")
+)
+
+type SidecarRenderer interface {
```

Comment:
i think it's might be better for this interface to just be like a mutatingwebhook.

like `mutate(pod *corev1.Pod) error` or `mutate(pod *corev1.pod) (*corev1.pod, error)`.
it's more flexiable.

Reply:
ok

Reply:
something like this
```
func NewVNEnvoyMutator(config, vn) ...
type VNEnvoyMutator struct {
    config EnvoyMutatorConfig
    vn appmesh.virtualNode
}

func (m *EnvoyMutator) Mutate(pod *corev1.pod) error {
    pod.annotation[virtualnode] = vn.name
}


func injectPod(pod, vn) error {
    mutators = []Mutator{
        NewVNEnvoyMutator(config, vn, ...)
        ...
    }
    for _, mutator := range mutators{
        mutator.mutate(pod)
    }
}
```